lib/cmdlib/cluster/__init__.py

```diff
@@ -1367,6 +1367,12 @@ def CheckPrereq(self):
             else:
               self.new_os_hvp[os_name][hv_name].update(hv_dict)
 
+      # Cleanup any OS that has an empty hypervisor parameter list, as we don't
+      # need them in the cluster config anymore.
+      for os_name, hvs in self.new_os_hvp.items():
+        if not hvs:
+          self.new_os_hvp.pop(os_name, None)
+
```

Comment:
It should not break any workflow because so far it was not possible to clean this file from stale/old OS images that were kept track of in the cluster config data. This change is just making sure that in case there are any, we properly get rid of them.

I'll be looking into the documentation to see if there are any places that I need to update about this change, but I couldn't find any on a cursory look. I don't think we document removing OSs or OS hypervisor parameters anywhere...